database/migrate_legacy_data_20220101.sh

```diff
@@ -0,0 +1,226 @@
+      ((count++))
+      # insert statements have a limit on the amount of values being inserted
+      # at once, this ensures that limit is never reached
+      if [[ "$count" -ge 100 ]]; then
+        rows="$(format_values_db 4 "${values[@]}")"
+        insert_into '"statistics_report"' "$columns" "$rows"
+
+        values=()
+        count=0
+      fi
+    done < "$file"
+  done
+
+  # insert_values '"statistics_report"' "$columns" 4 "${values[@]}"
+
+  # if there are no values left, return, otherwise insert them
+  [[ "${#values}" == 0 ]] && return
+
+  rows="$(format_values_db 4 "${values[@]}")"
+  insert_into '"statistics_report"' "$columns" "$rows"
+}
+
+function migrate_pomodoro()
+{
+  local -a file_list
+  local line
+  local tag
+  local start_date
+  local start_time
+  local duration
+  local description
+  local rows
+  local columns='("tag","date","time","duration","description")'
+  local count=0
+  local -a values=()
+
+  file_list=$(find "${datadir}/pomodoro" -type f | sort)
+
+  for file in $file_list; do
+    # avoid processing tags file
+    [[ "$file" =~ tags$ ]] && continue
+    # This line converts the file path into a usable date value
+    # E.g.: ${datadir}/pomodoro/2021/10/05 -> 2021-10-05
+    start_date=$(printf '%s\n' "$file" | sed -e 's/.*pomodoro\///' -e 's/\//-/g')
+    while read -r line; do
+      tag=$(printf '%s\n' "$line" | cut -d ',' -f1)
+      duration=$(printf '%s\n' "$line" | cut -d ',' -f2)
+      start_time=$(printf '%s\n' "$line" | cut -d ',' -f3)
+      description=$(printf '%s\n' "$line" | cut -d ',' -f1,2,3 --complement)
+
+      [[ -z "$description" ]] && description='NULL'
+      duration=$(timebox_to_sec "$duration")
+
+      values+=("$tag" "$start_date" "$start_time" "$duration" "$description")
+      ((count++))
+      # insert statements have a limit on the amount of values being inserted
+      # at once, this ensures that limit is never reached
+      if [[ "$count" -ge 100 ]]; then
+        rows="$(format_values_db 5 "${values[@]}")"
+        insert_into '"pomodoro_report"' "$columns" "$rows"
+
+        values=()
+        count=0
+      fi
+    done < "$file"
+  done
+
+  # insert_values '"pomodoro_report"' "$columns" 5 "${values[@]}"
+
+  # if there are no values left, return, otherwise insert them
+  [[ "${#values}" == 0 ]] && return
+
+  rows="$(format_values_db 5 "${values[@]}")"
+
+  insert_into '"pomodoro_report"' "$columns" "$rows"
+}
+
+function migrate_configs()
+{
+  local -a file_list
+  local line
+  local name
+  local path
+  local description
+  local rows
+  local columns='("name","description","path")'
+  local count=0
+  local -a values=()
+
+  local configs_dir="${datadir}/configs/configs"
+  local metadata_dir="${datadir}/configs/metadata"
+
+  file_list=$(find "$configs_dir" -type f | sort)
+
+  for file in $file_list; do
+    path="$file"
+    name="${path##*/}" # get just the file name
+    description=$(cat "${metadata_dir}/${name}")
```

Comment:
Check: https://kworkflow.org/content/codingstyle.html#command-substitution-and-arithmetic-expression